{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/local/lib/python2.7/dist-packages/ipykernel_launcher.py:12: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  if sys.path[0] == '':\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "from operator import itemgetter\n",
    "\n",
    "gene_coords_file = '/data/EBV/by_gene/gene_result.txt'\n",
    "repeats_analysis_file = '/home/ariel/Projects/Gutierrez/EBV-recomb/recomb/rdp4_results/ids_171_ebv/10Miter/results/repeats_analysis.csv'\n",
    "dfrep = pd.read_csv(repeats_analysis_file)\n",
    "\n",
    "gene_coords = pd.read_csv(gene_coords_file,sep = '\\t')\n",
    "gene_ref_coords = gene_coords[['GeneID','Symbol','Aliases','start_position_on_the_genomic_accession','end_position_on_the_genomic_accession']]\n",
    "gene_ref_coords['rang'] = gene_coords.apply(lambda x: (x.start_position_on_the_genomic_accession,\n",
    "                                                   x.end_position_on_the_genomic_accession),1)\n",
    "gene_ref_coords.set_index(['rang'],inplace = True)\n",
    "\n",
    "def look_for_gene(position,gene_coords):\n",
    "    res = gene_coords[(position> gene_coords.start_position_on_the_genomic_accession) & (position<gene_coords.end_position_on_the_genomic_accession)]\n",
    "    return(res.Symbol.unique())\n",
    "\n",
    "start_col = 'start_position_on_the_genomic_accession'\n",
    "end_col = 'end_position_on_the_genomic_accession'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "gen_st =dfrep.st.apply(lambda x:np.unique([look_for_gene(position=x,gene_coords=gene_ref_coords[gene_ref_coords.Symbol!='LMP2'])]))\n",
    "gen_end =dfrep.end.apply(lambda x:np.unique([look_for_gene(position=x,gene_coords=gene_ref_coords[gene_ref_coords.Symbol!='LMP2'])]))\n",
    "\n",
    "refs = []\n",
    "for i in range(len(gen_st)):\n",
    "    refs.append(np.unique(np.concatenate([gen_st[i],gen_end[i]])))\n",
    "#refs = pd.concat(refs)\n",
    "\n",
    "refs2 = [r if len(r)>0 else ['intr'] for r in refs ]\n",
    "a = []\n",
    "for i in range(len(refs2)):\n",
    "    l = len(refs2[i])\n",
    "    if l==1:\n",
    "        a.append(pd.Series(i,refs2[i]))\n",
    "    else:\n",
    "        a.append(pd.Series([i]*l,refs2[i]))\n",
    "\n",
    "a = pd.concat(a)\n",
    "a = a.to_frame().reset_index()\n",
    "a.columns = ['GENE','indice']\n",
    "\n",
    "b = pd.merge(dfrep,a,left_index=True,right_on='indice',how = 'left')\n",
    "rep_with_gene = b[['Mean Rho/bp.','+95% CI','-95% CI','call','GENE']]\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>+95% CI</th>\n",
       "      <th>-95% CI</th>\n",
       "      <th>Mean Rho/bp.</th>\n",
       "      <th>call</th>\n",
       "      <th>end</th>\n",
       "      <th>gene</th>\n",
       "      <th>locus_tag</th>\n",
       "      <th>note</th>\n",
       "      <th>rpt_family</th>\n",
       "      <th>rpt_type</th>\n",
       "      <th>rpt_unit_range</th>\n",
       "      <th>st</th>\n",
       "      <th>trimmed_end</th>\n",
       "      <th>trimmed_st</th>\n",
       "      <th>GENE</th>\n",
       "      <th>indice</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>0.04533</td>\n",
       "      <td>0.039273</td>\n",
       "      <td>0.04228</td>\n",
       "      <td>tandem</td>\n",
       "      <td>88043.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>87834.0</td>\n",
       "      <td>89774.0</td>\n",
       "      <td>89562.0</td>\n",
       "      <td>EBNA3B</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>0.04533</td>\n",
       "      <td>0.039273</td>\n",
       "      <td>0.04228</td>\n",
       "      <td>tandem</td>\n",
       "      <td>88043.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>87834.0</td>\n",
       "      <td>89774.0</td>\n",
       "      <td>89562.0</td>\n",
       "      <td>EBNA3C</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    +95% CI   -95% CI  Mean Rho/bp.    call      end gene locus_tag note  \\\n",
       "42  0.04533  0.039273       0.04228  tandem  88043.0  NaN       NaN  NaN   \n",
       "43  0.04533  0.039273       0.04228  tandem  88043.0  NaN       NaN  NaN   \n",
       "\n",
       "   rpt_family rpt_type rpt_unit_range       st  trimmed_end  trimmed_st  \\\n",
       "42        NaN      NaN            NaN  87834.0      89774.0     89562.0   \n",
       "43        NaN      NaN            NaN  87834.0      89774.0     89562.0   \n",
       "\n",
       "      GENE  indice  \n",
       "42  EBNA3B      27  \n",
       "43  EBNA3C      27  "
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "b[(b.call=='tandem') &(b['Mean Rho/bp.']>thr)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>+95% CI</th>\n",
       "      <th>-95% CI</th>\n",
       "      <th>Mean Rho/bp.</th>\n",
       "      <th>call</th>\n",
       "      <th>end</th>\n",
       "      <th>gene</th>\n",
       "      <th>locus_tag</th>\n",
       "      <th>note</th>\n",
       "      <th>rpt_family</th>\n",
       "      <th>rpt_type</th>\n",
       "      <th>rpt_unit_range</th>\n",
       "      <th>st</th>\n",
       "      <th>trimmed_end</th>\n",
       "      <th>trimmed_st</th>\n",
       "      <th>GENE</th>\n",
       "      <th>indice</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>0.01727</td>\n",
       "      <td>0.01386</td>\n",
       "      <td>0.01551</td>\n",
       "      <td>tandem</td>\n",
       "      <td>168134.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>168036.0</td>\n",
       "      <td>172990.0</td>\n",
       "      <td>172883.0</td>\n",
       "      <td>LMP1</td>\n",
       "      <td>35</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    +95% CI  -95% CI  Mean Rho/bp.    call       end gene locus_tag note  \\\n",
       "53  0.01727  0.01386       0.01551  tandem  168134.0  NaN       NaN  NaN   \n",
       "\n",
       "   rpt_family rpt_type rpt_unit_range        st  trimmed_end  trimmed_st  \\\n",
       "53        NaN      NaN            NaN  168036.0     172990.0    172883.0   \n",
       "\n",
       "    GENE  indice  \n",
       "53  LMP1      35  "
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "b[(b.call=='tandem') &(b.GENE =='LMP1')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "def mapp_and_compute_rho(df,start_col = 'st',end_col = 'end'):\n",
    "    df = df.dropna(subset=[start_col,end_col])\n",
    "    start_aux = [x if x in mapp_dict.keys() else (x +1 ) for x in df[start_col].values]\n",
    "    end_aux = [x if x in mapp_dict.keys() else (x -1 ) for x in df[end_col].values]\n",
    "\n",
    "\n",
    "    df['trimmed_st'] = get_items(mapp_dict,start_aux)\n",
    "    df['trimmed_end'] = get_items(mapp_dict,end_aux)\n",
    "\n",
    "    computed_rates = df.apply(lambda x: pd.concat([x,compute_local_obs(rrate,r1 = x.trimmed_st,r2 = x.trimmed_end)]),axis =1)\n",
    "    return df,computed_rates\n",
    "\n",
    "def get_items(d,keylist):\n",
    "    return itemgetter(*keylist)(d)\n",
    "\n",
    "\n",
    "def search_fasta(pattern, file_path):\n",
    "    st = []\n",
    "    end = []\n",
    "    pat =[]\n",
    "    reverse = str(Seq(pattern).reverse_complement())\n",
    "    \n",
    "    for patt in [pattern,reverse]:\n",
    "        for record in SeqIO.parse(open(file_path, \"rU\"), \"fasta\"):\n",
    "            chrom = record.id\n",
    "            for match in re.finditer(patt, str(record.seq)):\n",
    "                start_pos = match.start() + 1\n",
    "                end_pos = match.end() + 1\n",
    "                st.append(start_pos)\n",
    "                end.append(end_pos)\n",
    "                pat.append(patt)\n",
    "    df = pd.DataFrame({'st':st,'end':end,'pattern':pat,'call':pattern})\n",
    "    df = df[['st','end','pattern','call']]\n",
    "    df.sort_values(by=['st'],inplace = True)\n",
    "    \n",
    "    return(df)\n",
    "\n",
    "\n",
    "def compute_local_obs(rrate,r1,r2):\n",
    "    return(rrate[r1:r2].apply(np.mean))\n",
    "\n",
    "mapfile = '/data/EBV/msas/ids_171_ebv/ids_171_ebv_msa_gap1.51.fa_whole_mapping.file'\n",
    "outpath = '/home/ariel/Projects/Gutierrez/EBV-recomb/recomb/rdp4_results/ids_171_ebv/10Miter/results/'\n",
    "genbank_ref_file = '/data/EBV/byACCIDs/genBankRecord_NC_007605.gb'\n",
    "whole_recom_rate = '/home/ariel/Projects/Gutierrez/EBV-recomb/recomb/rdp4_results/ids_171_ebv/10Miter/whole_10Miters.tsv'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "####### map for all downstream analysis####\n",
    "###########################################\n",
    "mapp = pd.read_csv(mapfile,sep = '|')\n",
    "mapp_dict = mapp[['refCoord','trimmedCoord']]\n",
    "\n",
    "rrate = pd.read_csv(whole_recom_rate,sep = '|')\n",
    "rrate.set_index(['trimmed_position'],inplace = True)\n",
    "\n",
    "mapp_dict =  {}\n",
    "refcoords = mapp.loc[:,'refCoord']\n",
    "trimcoords = mapp.loc[:,'trimmedCoord']\n",
    "for i in range(mapp.shape[0]):\n",
    "    mapp_dict.update({refcoords[i]:trimcoords[i]})\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "def complete_features(seq_record_genbank, TIPO='regulatory',n=5 ):\n",
    "    #rates = res3[res3.type == TIPO]\n",
    "    #rates = rates[~rates['Mean Rho/bp.'].isnull()]\n",
    "\n",
    "    ttyp = []\n",
    "    loc = []\n",
    "    get = []\n",
    "    pos = []\n",
    "    for feature in seq_record.features:\n",
    "        ttyp.append(feature.type)\n",
    "\n",
    "        if TIPO[0:n] in feature.type:\n",
    "            get.append({k: feature.qualifiers[k][0] for k in feature.qualifiers.keys()})\n",
    "\n",
    "            partes = []\n",
    "            for p in feature.location.parts:\n",
    "                partes.append(pd.Series([p.start.position,p.end.position]))\n",
    "            positions = pd.concat(partes,1).transpose()    \n",
    "            positions.columns = ['start','end']\n",
    "            pos.append(positions)        #print feature\n",
    "\n",
    "    get = pd.DataFrame(get)        \n",
    "    pos = pd.concat(pos)\n",
    "\n",
    "#    return(get,pos)\n",
    "\n",
    "    pos.index = get.index\n",
    "    get_features = pd.concat([get,pos],1)\n",
    "        \n",
    "    #rho = rates[['gene','Mean Rho/bp.','start']]#'+95% CI','start','end']]\n",
    "    #rho = pd.merge(rho,get_features.drop(['gene'],1),on='start',how = 'outer')\n",
    "    return(get_features)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import SeqIO\n",
    "genbank_ref_file = '/data/EBV/byACCIDs/genBankRecord_NC_007605.gb'\n",
    "seq_record = SeqIO.read(genbank_ref_file,'genbank')\n",
    "\n",
    "types = []\n",
    "cds = []\n",
    "region =[]\n",
    "for feature in seq_record.features:\n",
    "    types.append(feature.type)\n",
    "    if 'gene' in feature.qualifiers.keys():\n",
    "        gene = feature.qualifiers['gene']\n",
    "    else:\n",
    "        gene = 'None'\n",
    "    \n",
    "    partes = []\n",
    "    for p in feature.location.parts:\n",
    "        partes.append(pd.Series([p.start.position,p.end.position]))\n",
    "    positions = pd.concat(partes,1).transpose()    \n",
    "    positions.columns = ['start','end']\n",
    "    positions['type'] = feature.type\n",
    "    positions['gene'] = np.repeat(gene,positions.shape[0])\n",
    "\n",
    "    \n",
    "    region.append(positions)\n",
    "regions = pd.concat(region)\n",
    "\n",
    "regions = regions[regions.type!='source']\n",
    "\n",
    "# compute length of each region\n",
    "regions['length'] = regions.end - regions.start\n",
    "regions['length'] = pd.to_numeric(regions.length)\n",
    "\n",
    "regions = regions[regions.length < 40000] # con esto tiramos la anomalia del primer gen. \n",
    "regions.rename(columns={'gene':'GENE','type':'call','start':'st'},inplace = True)\n",
    "_, regions_rho = mapp_and_compute_rho(regions)\n",
    "reg_rho = regions_rho[['call','GENE','Mean Rho/bp.','-95% CI','+95% CI']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "reg_rho = regions_rho[['call','GENE','Mean Rho/bp.','-95% CI','+95% CI']]\n",
    "bygene = pd.concat([reg_rho,rep_with_gene])\n",
    "bygene.dropna(subset = ['Mean Rho/bp.'],inplace = True)\n",
    "result = bygene.groupby(['GENE','call'])['Mean Rho/bp.'].mean()\n",
    "result = result.reset_index()\n",
    "result.replace({'intr':'LMP2'},inplace = True)  # PARCHE\n",
    "result.call.replace('','NC.',inplace = True)\n",
    "#result.to_csv(outpath+'gene_vs_call.csv')\n",
    "#result.to_csv(outpath+'gene_vs_call.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "28\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/local/lib/python2.7/dist-packages/pandas/core/generic.py:3924: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  self._update_inplace(new_data)\n"
     ]
    }
   ],
   "source": [
    "thr = rrate.mean()['-95% CI']\n",
    "genes_ok = result[result['Mean Rho/bp.']>thr]['GENE']\n",
    "print genes_ok.nunique()\n",
    "result_above_mean= result[result.GENE.isin(genes_ok)]\n",
    "result_above_mean.call.replace('','NC.',inplace = True)\n",
    "\n",
    "#result_above_mean.to_csv(outpath+'filtered_gene_vs_call.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GENE</th>\n",
       "      <th>call</th>\n",
       "      <th>Mean Rho/bp.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>176</th>\n",
       "      <td>EBNA3B</td>\n",
       "      <td>tandem</td>\n",
       "      <td>0.021079</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>181</th>\n",
       "      <td>EBNA3C</td>\n",
       "      <td>tandem</td>\n",
       "      <td>0.028925</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>222</th>\n",
       "      <td>LMP2</td>\n",
       "      <td>TERMINAL</td>\n",
       "      <td>0.000010</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>223</th>\n",
       "      <td>LMP2</td>\n",
       "      <td>type A</td>\n",
       "      <td>0.068306</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>224</th>\n",
       "      <td>LMP2</td>\n",
       "      <td>type B</td>\n",
       "      <td>0.283990</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>225</th>\n",
       "      <td>LMP2</td>\n",
       "      <td>type C</td>\n",
       "      <td>0.304569</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>226</th>\n",
       "      <td>LMP2</td>\n",
       "      <td>type D</td>\n",
       "      <td>0.465500</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       GENE      call  Mean Rho/bp.\n",
       "176  EBNA3B    tandem      0.021079\n",
       "181  EBNA3C    tandem      0.028925\n",
       "222    LMP2  TERMINAL      0.000010\n",
       "223    LMP2    type A      0.068306\n",
       "224    LMP2    type B      0.283990\n",
       "225    LMP2    type C      0.304569\n",
       "226    LMP2    type D      0.465500"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "thr = rrate.mean()['-95% CI']\n",
    "rrep = ['inverted', 'tandem', 'nan', 'type A', 'type B', 'type C', 'NC.','type D','EBNA triplet repeat GGA,GCA,GGG','PstI repeats', 'TERMINAL']\n",
    "repet = result[result.call.isin(rrep)]\n",
    "genes_ok = repet[repet['Mean Rho/bp.']>thr]['GENE']\n",
    "print genes_ok.nunique()\n",
    "\n",
    "result_above_mean= repet[repet.GENE.isin(genes_ok)]\n",
    "#result_above_mean.call.replace('','NC.',inplace = True)\n",
    "#result_above_mean.to_csv(outpath+'filtered_gene_vs_repeat.csv')\n",
    "result_above_mean"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Mean Rho/bp.</th>\n",
       "      <th>+95% CI</th>\n",
       "      <th>-95% CI</th>\n",
       "      <th>call</th>\n",
       "      <th>GENE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>inverted</td>\n",
       "      <td>intr</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>inverted</td>\n",
       "      <td>intr</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>inverted</td>\n",
       "      <td>intr</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>inverted</td>\n",
       "      <td>intr</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>inverted</td>\n",
       "      <td>BCRF2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Mean Rho/bp.  +95% CI  -95% CI      call   GENE\n",
       "0           NaN      NaN      NaN  inverted   intr\n",
       "1           NaN      NaN      NaN  inverted   intr\n",
       "2           NaN      NaN      NaN  inverted   intr\n",
       "3           NaN      NaN      NaN  inverted   intr\n",
       "4           NaN      NaN      NaN  inverted  BCRF2"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rep_with_gene.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "gene             151\n",
       "CDS              138\n",
       "regulatory       125\n",
       "misc_feature      51\n",
       "exon              39\n",
       "repeat_region     39\n",
       "mRNA              24\n",
       "intron             4\n",
       "variation          4\n",
       "rep_origin         4\n",
       "misc_RNA           3\n",
       "ncRNA              3\n",
       "STS                3\n",
       "mat_peptide        2\n",
       "Name: call, dtype: int64"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "regions_rho.call.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
